calyx-ir/src/guard.rs

```diff
@@ -1,8 +1,19 @@
+use super::Group;
 use super::{Port, RRC};
+use std::fmt::Debug;
 use std::mem;
 use std::ops::{BitAnd, BitAndAssign, BitOr, BitOrAssign, Not};
 use std::{cmp::Ordering, hash::Hash, rc::Rc};
 
+#[derive(Debug, Clone, Default)]
+pub struct Nothing;
```

Comment:
I'm curious why this is needed. Can we use `()` instead? If the pretty printing is causing problem, you can specifically specify it by doing:
```
impl Format for Guard<()> { ... }
```